### PR TITLE
Add preliminary support for parsing of command line options

### PR DESCRIPTION
**Pull request by Sword-Smith**
This looks like an excellent addition by @lennonhill. Merge?

### CODE REVIEW

File: src/Mapscii.js [hunk 1]
```diff
@@ -101,8 +97,8 @@ class Mapscii {
       this.width = config.size.width;
       this.height = config.size.height;
     } else {
-      this.width = config.output.columns >> 1 << 2;
-      this.height = config.output.rows * 4 - 4;
+      this.width = config.width ? config.width : config.output.columns >> 1 << 2;
```
Comment (jaller94) [inline src/Mapscii.js:100]: Why introduce `config.width` and `config.height` if there's already `config.size` with both these values?

Comment (lennonhill) [reply src/Mapscii.js:100]: This was intended as a preliminary way to expose some options via command line parameters, including width and height. As they are nested via the size structure, a (simple) parser would be required to set these. I don't have any experience with node, so I don't know how/where to handle parsing of command line arguments in this case.

Comment (jaller94) [reply src/Mapscii.js]: All args will need to be validated anyway. Those numbers will need to be parsed to the number type as well.
As part of these validations and transformations we can have them override the values from the config.

**Comment by lennonhill** [comment]
Type validation is still outstanding, as the argument parser is not strict about those.

File: src/Mapscii.js (hunk 1 above)
Comment (rastapasta) [reply src/Mapscii.js]: Setting width and height manually will interfere with autoresizing - implying: if width/height is given as argv, autoresize should be disabled.

**Comment by rastapasta** [comment]
Aloha and thanks for checkin' in :) I would strongly prefer using ```yargs```'s way of defining default values instead of having to merge objects and filtering by unset args. Feels a bit error-prone. Values can still be pulled in from the config, it should just be avoided that the variables are set at different points at code @ maintainability.

File: src/Mapscii.js (hunk 1 above)
Comment (lennonhill) [reply src/Mapscii.js]: Does https://github.com/rastapasta/mapscii/blob/393f867c847f5789947347f0016df463aae3cfcf/src/Mapscii.js#L86-L106 not effectively disable autoresizing?

**Comment by lennonhill** [comment]
Is this better?

File: src/config.js [hunk 2]
```diff
@@ -13,6 +13,13 @@ module.exports = {
   maxZoom: 18,
   zoomStep: 0.2,
 
+  // sf lat: 37.787946, lon: -122.407522
+  // iceland lat: 64.124229, lon: -21.811552
+  // rgbg
+  // lat: 49.019493, lon: 12.098341
+  initialLat: 52.51298,
+  initialLon: 14.42012,
```
Comment (jaller94) [inline src/config.js:21]: Point this back at the c-base in Berlin.
```suggestion
  initialLat: 52.51298,
  initialLon: 13.42012,
```

**Pull request merged**